src/matrix/room/timeline/Timeline.js

```diff
@@ -236,8 +247,50 @@ export class Timeline {
     /** @package */
     addEntries(newEntries) {
         this._addLocalRelationsToNewRemoteEntries(newEntries);
+        this._updateFetchedEntries(newEntries);
+        this._loadRelatedEvents(newEntries);
         this._remoteEntries.setManySorted(newEntries);
     }
+
+    _updateFetchedEntries(entries) {
+        for (const entry of entries) {
+            const relatedEntry = this._fetchedEventEntries.find(e => e.id === entry.relatedEventId);
+            if (relatedEntry?.addLocalRelation(entry)) {
```

Comment:
this is to make fetched event appear as redacted (and later edited) when we get a redaction? perhaps add a comment saying as much.